src/ott/solvers/linear/univariate.py

```diff
@@ -0,0 +1,103 @@
+# Copyright OTT-JAX
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from typing import Literal, Optional
+
+import jax
+import jax.numpy as jnp
+
+from ott.geometry import costs
+from ott.tools import soft_sort
+
+__all__ = ["UnivariateSolver"]
+
+
+@jax.tree_util.register_pytree_node_class
+class UnivariateSolver:
+  """1-D optimal transport solver.
+
+  warning::
+  As implemented, this solver assumes uniform marginals,
+  non-uniform marginal solver coming soon!
+
+  Computes the 1-Dimensional Optimal Transport distance between two histograms
+  via a variety of methods.
+
+  Args:
+    epsilon_sort: regularization parameter for sorting. 0.0 for hard-sorting.
+    cost_fn: The cost function for transport. Defaults to Squared Euclidean
+    distance.
+    method: The method used for computing the distance on the line. Options
+    currently supported are:
+      subsample: Take a stratfied subsample of the distances,
+      quantile: Take equally spaced quantiles of the distances,
+      equal: No subsampling is performed--requires distributions to have the
+      same number of points.
+    n_subsamples: The number of subsamples to draw for the "quantile" or
+    "subsample" methods
+    require_sort: Whether to assume that the inputed arrays are sorted.
+  """
+
+  def __init__(
+      self,
+      epsilon_sort: Optional[float] = 0.0,
+      cost_fn: Optional[costs.TICost] = None,
+      method: Literal["subsample", "quantile", "equal"] = "subsample",
+      n_subsamples: int = 100,
```

Comment:
Am pondering whether this should have a default or not (None), wdyt @marcocuturi ?